### DISCUSSION TITLE
Gradle and Maven resolving different native bindings for netty-transport-native-unix-common

### DISCUSSION THREAD

**Discussion by matthisk** [netty/netty#13847] [Q&A]
Hello!

I have two projects set up locally which both (transitively) depend on netty-transport-native-unix-common. My architecture is Darwin (m2). One project is managed through Maven the other through Gradle. When I first resolve and cache the dependencies for the maven project the `netty-transport-native-unix-common-4.1.101.Final.jar` is downloaded and stored in m2 cache. When I then try to resolve the dependencies for the Gradle project, this fails because it looks for `netty-transport-native-unix-common-4.1.101.Final-linux-x86_64.jar` which it cannot find in the `.m2` repository. 

Is this an issue with Maven, Gradle, or with the netty unix bindings? Any help is much appreciated :)

**Comment by slandelle** [comment]
https://github.com/gradle/gradle/issues/21132 ?

**Reply by matthisk** [reply]
Maven doesn't resolve any classifiers though. It is Gradle that tries to resolve the Linux x86 classifier, and fails because maven didn't resolve this variant into its local repository?

(to be honest, I have limited experience with the resolution of these JNI variants of packages; so I might be missing something)

**Reply by slandelle** [reply]
Maven has it right. Gradle's support of classifiers is broken.

**Reply by chrisvest** [reply]
In gradle I think you have to just explicitly depend on the transitive dependencies with the classifiers you want.
